client/lib/domains/get-domains-eligible-to-purchase-email.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { canCurrentUserAddEmail } from './can-current-user-add-email';
+import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+
+/**
+ * Filters a list of domain objects by their eligibility to purchase email.
+ *
+ * - This excludes domains that already have email subscriptions and then checks if the rest can add email
+ *
+ * @param {Array} domains - list of domain objects
+ * @returns {Array} - the resultant list of eligible domains objects
+ */
+export function getDomainsEligibleToPurchaseEmail( domains ) {
+	return domains.filter( ( domain ) => {
+		if ( hasGSuiteWithUs( domain ) || hasTitanMailWithUs( domain ) ) {
+			return false;
+		}
+
+		return canCurrentUserAddEmail( domain );
+	} );
+}
```

Comment:
@stephanethomas, I don't know that we should add the existing service check to the back end logic -- that would make it impossible to switch between our paid email services.

I would also be wary of adding any provider-specific logic to that class because it is currently provider-independent and it's output is currently a boolean that applies to all email services.

Reply:
I am with @daledupreez on this. I think we should keep `Domain_Email_Eligibility` provider agnostic.

> We are also introducing a new function that is almost similar in purpose to `canCurrentUserAddEmail`

I admit that the functionalities are similar, but the concepts are distinct. Mostly the `canCurrentUserAddEmail` features the result of the `Domain_Email_Eligibility` class which does not consider existing subscriptions, so we need to factor this in now.

> 1. We no longer check if the domain is hosted on WordPress.com for Google Workspace

Since we just want to figure out if the user can add either type of paid email service, we no longer need to check if the domain is hosted on WordPress.com because Professional email does not have that restriction. We don't know what the user prefers at this point. Their choices can be further whittled down at the Email Comparison page.

> 2. We no longer check if the domain is an Atomic staging site for Google Workspace

`Domain_Management_Domain` already covers this checks at the backend. Subdomains and Atomic staging domains are already excluded.